guides/ADDING_A_NEW_PRODUCT.md

```diff
@@ -0,0 +1,160 @@
+all the logic in here should be written with the assumption that resources may or may not exist, and can
+be created or updated based on their current state. 
+
+As a side-note, it's worth keeping in mind that there are quite a lot of helpful functions in the resource package for 
+handling the more common tasks, such as setting up the CSV, catalog source, subscription and namespaces. Many examples 
+of this are available in the other reconcilers (such as Code-Ready, RHSSO, etc). 
+
+There is also a helper function to add a finalizer to the RHMI CR, this also requires a function allowing the organisation 
+of the tear-down of the product at uninstall time, again examples of this can are available in the existing reconcilers.
+
+#### Parameters
+
+There are several parameters passed into this function:
+##### ctx
+ This must be used in all network requests performed by the reconciler, as the integreatly-operator maintains this context
+ and may kill it if an uninstall is detected.
+ 
+##### installation 
+This is the CR we are basing the install from, it has values that are occasionally required by reconcilers, for example
+the namespace prefix. 
+
+##### product 
+This is a pointer to the this reconciler's product in the status block of the CR, it can be used to set values
+such as version, host and operator version.
+
+##### serverClient
+This is the client to the cluster, and is used for getting, creating, updating and deleting resources in the cluster.
+
+#### Return Values
+The return values from this method are `state` and `err`:
+##### State
+This is communicated back to the user via the status block of the RHMI CR, this is usually either in progress
+or complete. It can go to `fail` if something has broken, but this will not prevent the installation_controller
+from calling the Reconcile function in the future, which may allow the reconciler to fix whatever issue had
+occurred (i.e. the service had not come up yet, so there were network errors accessing it's API).
+
+##### Err 
+This is how we can communicate to the user via the status block of the RHMI CR what is causing a product to 
+enter a failed state, and is written into the `status.lastError` of the CR along with any other errors from
+other reconcilers.
+
+### GetPreflightObject
+Before the integreatly-operator will begin an installation, it will initially check the cluster has no existing installs
+of the same products; this is to avoid potential issues of 2 operators trying to act on one resource. 
+
+This function informs the operator of what object it should look for, to check if the product is already installed. The 
+namespace argument is the namespace currently being scanned for existing installations.
+
+For example, codeready looks for a deployment in the scanned namespace with the name "codeready", if found this 
+installation will stall until that product is removed.
+
+## Update Reconciler Factory
```

Comment:
Would be good to put a link to this and exactly what needs to be updated.

Reply:
Good thinking, I've added quite a few links to stuff now.